Tests/SentryTests/Integrations/ANR/SentryANRTrackingIntegrationTests.swift

```diff
@@ -68,38 +52,45 @@ class SentryANRTrackingIntegrationTests: XCTestCase {
         assertArrayEquals(expected: expexted, actual: Array(options.enabledIntegrations))
     }
     
-    func testANRDetected_UpdatesAppStateToTrue() {
+    func testANRDetected_EventCaptured() {
         givenInitializedTracker()
         
         Dynamic(sut).anrDetected()
         
-        guard let appState = fixture.fileManager.readAppState() else {
-            XCTFail("appState must not be nil")
-            return
-        }
-
-        XCTAssertTrue(appState.isANROngoing)
-    }
-    
-    func testANRStopped_UpdatesAppStateToFalse() {
-        givenInitializedTracker()
-        
-        Dynamic(sut).anrStopped()
-        
-        guard let appState = fixture.fileManager.readAppState() else {
-            XCTFail("appState must not be nil")
-            return
+        assertEventWithScopeCaptured { event, _, _ in
+            XCTAssertNotNil(event)
+            guard let ex = event?.exceptions?.first else {
+                XCTFail("ANR Exception not found")
+                return
+            }
+            
+            XCTAssertEqual(ex.mechanism?.type, "anr")
+            XCTAssertEqual(ex.type, "App Hanging")
+            XCTAssertEqual(ex.value, "Application Not Responding for at least 4500 ms.")
+            
+            guard let threads = event?.threads else {
+                XCTFail("ANR Exception not found")
+                return
+            }
+            
+            //Sometimes during tests its possible to have one thread without frames
+            //We just need to make sure we retrieve frame information for at least one other thread than the main thread
+            var threadsWithFrames = 0
+            
+            for thr in threads {
+                if (thr.stacktrace?.frames.count ?? 0) >= 1 {
+                    threadsWithFrames += 1
+                }
+            }
```

Comment:
`l`: A bit cleaner maybe
```suggestion
            let threadsWithFrames = threads.filter {
                ($0.stacktrace?.frames.count ?? 0) >= 1
            }.count
```